packages/browser-wallet/src/popup/page-layouts/MainLayout/AccountList/AccountList.tsx

```diff
@@ -1,21 +1,83 @@
-import React, { forwardRef, useMemo } from 'react';
+import React, { forwardRef, KeyboardEventHandler, useCallback, useMemo, useState } from 'react';
 import clsx from 'clsx';
-import { useAtomValue, useAtom } from 'jotai';
+import { useAtom, useAtomValue } from 'jotai';
 import { useNavigate } from 'react-router-dom';
 import { ClassName, displayAsCcd } from 'wallet-common-helpers';
 import CopyButton from '@popup/shared/CopyButton';
 import CheckmarkIcon from '@assets/svg/checkmark-blue.svg';
+import EditIcon from '@assets/svg/edit-secondary.svg';
 import BakerIcon from '@assets/svg/validator.svg';
 import DelegationIcon from '@assets/svg/delegation.svg';
 import { absoluteRoutes } from '@popup/constants/routes';
 import { credentialsAtom, selectedAccountAtom } from '@popup/store/account';
 import { useTranslation } from 'react-i18next';
-import { displaySplitAddress, useIdentityName } from '@popup/shared/utils/account-helpers';
+import {
+    displayNameOrSplitAddress,
+    useIdentityName,
+    useWritableSelectedAccount,
+} from '@popup/shared/utils/account-helpers';
 import { WalletCredential } from '@shared/storage/types';
 import { useAccountInfo } from '@popup/shared/AccountInfoListenerContext';
-import { isDelegatorAccount, isBakerAccount, AccountInfo } from '@concordium/web-sdk';
+import { AccountInfo, isBakerAccount, isDelegatorAccount } from '@concordium/web-sdk';
+import IconButton from '@popup/shared/IconButton';
+import { InlineInput } from '@popup/shared/Form/InlineInput';
 import EntityList from '../EntityList';
 
+const ACCOUNT_NAME_MAX_LENGTH = 12;
```

Comment:
What's the reason behind this max length? I could see people wanting to name their accounts something longer than this 🙂

Reply:
Places where the account name is used, have been designed for the shortened account address. With this limitation it won't break existing design.